packages/core/test/integration/dialects/mssql/query-queue.test.ts

```diff
@@ -0,0 +1,89 @@
+import { expect } from 'chai';
+import type { CreationOptional, InferAttributes, InferCreationAttributes } from '@sequelize/core';
+import { AsyncQueueError, ConnectionError, DataTypes, Model } from '@sequelize/core';
+import type { MsSqlConnection } from '@sequelize/core/_non-semver-use-at-your-own-risk_/dialects/mssql/connection-manager';
+import { Attribute, NotNull } from '@sequelize/core/decorators-legacy';
+import { beforeAll2, sequelize, setResetMode } from '../../support';
+
+describe('[MSSQL Specific] Query Queue', () => {
+  if (!sequelize.dialect.name.startsWith('mssql')) {
+    return;
+  }
+
+  setResetMode('none');
+
+  const vars = beforeAll2(async () => {
+    class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
+      declare id: CreationOptional<number>;
+
+      @Attribute(DataTypes.STRING)
+      @NotNull
+      declare username: string;
+    }
+
+    sequelize.addModels([User]);
+    await sequelize.sync({ force: true });
+    await User.create({ username: 'John' });
+
+    return { User };
+  });
+
+  it('should queue concurrent requests to a connection', async () => {
+    await expect(sequelize.transaction(async t => {
+      return Promise.all([
+        vars.User.findOne({ transaction: t }),
+        vars.User.findOne({ transaction: t }),
+      ]);
+    })).not.to.be.rejected;
+  });
+
+  it('requests that reject should not affect future requests', async () => {
+    await expect(sequelize.transaction(async t => {
+      // @ts-expect-error -- testings invalid input
+      await expect(vars.User.create({ username: new Date() })).to.be.rejected;
+      await expect(vars.User.findOne({ transaction: t })).not.to.be.rejected;
+    })).not.to.be.rejected;
+  });
+
+  it('closing the connection should reject pending requests', async () => {
```

Comment:
Just to check; in this test and the test below you now check that the cause is AsyncQueueError instead of the parent. Correct? Or did you change more for this to work?

Reply:
I made this change as `parent` has been deprecated as per https://github.com/sequelize/sequelize/blob/d2581e0c5d3cdf443974568d523eefa429c313d7/packages/core/src/errors/base-error.ts#L35-L42